### PR TITLE
Redesign buffer data storage

### PR DESCRIPTION
**Pull request by Benjamin-Philip**
- Add from_erlang/2 to serde_arrow_buffer
- Add to_arrow/2

### CODE REVIEW

**Comment by Benjamin-Philip** [comment]
I've decided to not serialize data in the intermediate format, and only at the `to_arrow` stage. I've also decided to add 4 new functions to replace `new/2` (which is what will happen to the array layouts):

- `from_erlang/2`
- `to_arrow/1`
- `from_arrow` (Arity not fixed)
- `to_erlang/1`

File: src/serde_arrow_buffer.erl [hunk 1]
```diff
@@ -53,6 +53,44 @@ new(Values, Type) ->
...
+                    end,
+                    0,
+                    Data
+                );
+            _ ->
+                length(Data) * serde_arrow_type:byte_length(Type)
+        end,
+    #buffer{type = Type, length = Len, data = Data}.
+
+%% @doc God knows
```
Comment (josevalim) [inline src/serde_arrow_buffer.erl:83]: :D

**Review by josevalim — APPROVED** [review]

File: src/serde_arrow_buffer.erl (hunk 1 above)
Comment (Benjamin-Philip) [reply src/serde_arrow_buffer.erl:83]: I forgot to commit the docs. That was simply some placeholder text to remind me to write it up 😅

**Comment by Benjamin-Philip** [comment]
I'll make a separate PR for `from_arrow` later. There are a number issues that need to be dealt with before I can work on it (like redesign of the internal representation of validity bitmaps), plus it is not urgent at the project's s present state.

**Review by josevalim — APPROVED** [review]

**Pull request merged**